watertap/property_models/unit_specific/activated_sludge/modified_asm2d_properties.py

```diff
@@ -442,10 +522,90 @@ def _TSS(self):
         )
 
         def rule_TSS(b):
-            return b.TSS == b.VSS + b.ISS
+            return b.TSS == b.conc_mass_comp["X_H"]
 
         self.eq_TSS = pyo.Constraint(rule=rule_TSS)
 
+    def _COD(self):
+        self.COD = pyo.Var(
+            initialize=1,
+            domain=pyo.NonNegativeReals,
+            doc="Chemical oxygen demand",
+            units=pyo.units.kg / pyo.units.m**3,
+        )
+
+        def rule_COD(b):
+            return (
+                b.COD
+                == b.conc_mass_comp["S_F"]
+                + b.conc_mass_comp["S_A"]
+                + b.conc_mass_comp["S_I"]
+                + b.conc_mass_comp["X_I"]
+                + b.conc_mass_comp["X_S"]
+                + b.conc_mass_comp["X_H"]
+                + b.conc_mass_comp["X_PAO"]
+                + b.conc_mass_comp["X_PHA"]
+                + b.conc_mass_comp["X_AUT"]
+            )
+
+        self.eq_COD = pyo.Constraint(rule=rule_COD)
+
+        def rule_SNKj(b):
+            return b.SNKj == b.conc_mass_comp[
+                "S_NH4"
+            ] + b.params.i_NSF * b.conc_mass_comp[
+                "S_F"
+            ] + b.params.i_NSI * b.conc_mass_comp[
+                "S_I"
+            ] + b.params.i_NXI * b.conc_mass_comp[
+                "X_I"
+            ] + b.params.i_NXS * b.conc_mass_comp[
+                "X_S"
+            ] + b.params.i_NBM * (
+                b.conc_mass_comp["X_H"]
+                + b.conc_mass_comp["X_PAO"]
+                + b.conc_mass_comp["X_AUT"]
+            )
+
+        self.eq_SNKj = pyo.Constraint(rule=rule_SNKj)
+
+        def rule_BOD5(b):
+            return b.BOD5 == 0.25 * (
```

Comment:
From memory, I recall that the factor (0.25) could change depending on whether this is calculated for influent or effluent. Can you double-check this? I handled this in some way for ASM1 (though it might not be ideal approach).

Reply:
I am familiar with what you are talking about, but the c-code and the literature reference both hard-code 0.25, so I didn't see a need to do what you did in ASM1

Reply:
The problem here is if I want to apply to the influent, that value changes, right? Since this can be constructed on any state block and not only one that corresponds to treated effluent, we cannot just hard-code the value to what would correspond to the effluent.

Reply:
Can you show the same snapshot for BOD5i or whatever corresponds to influent in that file?

Reply:
Ah yes, the influent value is 0.65
![image](https://github.com/user-attachments/assets/d195f16e-d928-43b5-8fe1-077cf14c6064)